plugins/module_utils/ilo_redfish_utils.py

```diff
@@ -228,3 +229,79 @@ def set_wins_registration(self, mgrattr):
         if not response['ret']:
             return response
         return {'ret': True, 'changed': True, 'msg': "Modified %s" % mgrattr['mgr_attr_name']}
+
+    def get_server_poststate(self):
+        # Get server details
+        response = self.get_request(self.root_uri + self.systems_uri)
+        if not response["ret"]:
+            return response
+        server_data = response["data"]
+
+        if "Hpe" in server_data["Oem"]:
+            return {
+                "ret": True,
+                "server_poststate": server_data["Oem"]["Hpe"]["PostState"]
+            }
+        else:
+            return {
+                "ret": True,
+                "server_poststate": server_data["Oem"]["Hp"]["PostState"]
+            }
+
+    def check_ilo_reboot_status(self, polling_interval=60, max_polling_time=1800):
```

Comment:
Is there a reason this needs to be polling for a final state? To me, when I see a command like "check reboot status", I expect it to simply return the status. If the desire is to wait until a final state, maybe a different name like "wait_for_ilo_reboot_completion" would be better.

Reply:
Agreed, changed function name to **"wait_for_ilo_reboot_completion"** and command to **"WaitforiLORebootCompletion"**.

Reply:
Resolving this conversation. Thanks!